File: clients/python/examples/quickstart.py
"""PolarGraph Python SDK — quickstart example.

Run against a local polargraphd:
    python examples/quickstart.py
"""

import uuid
from polargraph import PolarGraphClient

alice_id = str(uuid.uuid4())
bob_id   = str(uuid.uuid4())
carol_id = str(uuid.uuid4())

with PolarGraphClient("localhost", 50051) as client:
    # Insert three nodes
    client.insert_node(alice_id, "Person", name="Alice", age=30)
    client.insert_node(bob_id,   "Person", name="Bob",   age=25)
    client.insert_node(carol_id, "Person", name="Carol", age=28)

    # Insert two edges
    client.insert_edge(alice_id, "knows", bob_id,   since=2020)
    client.insert_edge(bob_id,   "knows", carol_id, since=2022)

    # Pattern query — who does Alice know?
    results = client.query([
        {"s": alice_id, "p": "knows", "o": "?friend"},
    ])
    print("Alice knows:")
    for row in results:
        print(" ", row)

    # Cypher query
    rows = client.cypher(
        'MATCH (a:Person)-[:knows]->(b:Person) RETURN a, b LIMIT 10'
    )
    print("\nCypher MATCH knows:")
    for row in rows:
        print(" ", row)

    # Streaming query
    print("\nStreaming all Person nodes:")
    for row in client.stream_query([{"s": "?n", "p": "__type", "o": None}]):
        print(" ", row)

File: clients/js/examples/quickstart.ts
/**
 * PolarGraph JavaScript SDK — quickstart example.
 *
 * Run against a local polargraphd:
 *   npx ts-node examples/quickstart.ts
 *
 * Or compile first:
 *   npm run build && node dist/cjs/examples/quickstart.js
 */

import { randomUUID } from "crypto";
import { PolarGraphClient } from "../src/index.js";

async function main() {
  const client = new PolarGraphClient("localhost", 50051);

  const aliceId = randomUUID();
  const bobId   = randomUUID();
  const carolId = randomUUID();

  // Insert three Person nodes
  await client.insertNode(aliceId, "Person", { name: "Alice", age: 30 });
  await client.insertNode(bobId,   "Person", { name: "Bob",   age: 25 });
  await client.insertNode(carolId, "Person", { name: "Carol", age: 28 });
  console.log("Inserted 3 nodes");

  // Insert two edges
  await client.insertEdge(aliceId, "knows", bobId,   { since: 2020 });
  await client.insertEdge(bobId,   "knows", carolId, { since: 2022 });
  console.log("Inserted 2 edges");

  // Pattern query — who does Alice know?
  const friends = await client.query([
    { s: aliceId, p: "knows", o: "?friend" },
  ]);
  console.log("\nAlice knows (pattern query):");
  for (const row of friends) {
    console.log("  friend =", row.friend);
  }

  // Cypher read query
  const cypherRows = await client.cypher(
    "MATCH (a:Person)-[:knows]->(b:Person) RETURN a, b LIMIT 10"
  );
  console.log("\nCypher MATCH knows:");
  for (const row of cypherRows) {
    console.log(" ", row);
  }

  // Cypher write
  const result = await client.cypherWrite(
    'CREATE (c:Company {name: "Acme"})'
  );
  console.log("\nCreated company node IDs:", result.createdNodeIds);
  console.log("Triples written:", result.triplesWritten);

  // Wire transaction example
  const txId = await client.beginTx();
  await client.insertNode(randomUUID(), "Person", { name: "Dave" });
  const triplesWritten = await client.commitTx(txId);
  console.log(`\nTransaction committed (${triplesWritten} triples)`);

  client.close();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
